framework/src/wifi_manager/wpa_wifi_utils.c

```diff
@@ -309,13 +309,18 @@ wifi_utils_result_e wifi_utils_deinit(void)
 wifi_utils_result_e wifi_utils_scan_ap(void *arg)
 {
 	wifi_utils_result_e wuret = WIFI_UTILS_FAIL;
-	int8_t ret = WiFiScanNetwork();
+	int8_t ret = WiFiRegisterScanCallback(&wifi_scan_result_callback);
```

Comment:
how about check wifi_scan_result_callback is null ?
if wifi_scan_result_callback points always a valid function address. you don't need to check

Reply:
Sorry I misunderstood. we don't need to check wifi_scan_result_callback.